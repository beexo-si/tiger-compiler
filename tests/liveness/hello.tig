/*TIGER_ARGS=-code -flow*/
(print ("Hello World");0)
