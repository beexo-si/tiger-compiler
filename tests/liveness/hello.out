bien!
;;--FRAME--L0__tigermain_0:
LABEL: L3
MOVE: movq %'s0, %'d0 D:T2 S:rbx
MOVE: movq %'s0, %'d0 D:T3 S:r12
MOVE: movq %'s0, %'d0 D:T4 S:r13
MOVE: movq %'s0, %'d0 D:T5 S:r14
MOVE: movq %'s0, %'d0 D:T6 S:r15
MOVE: movq %'s0, %'d0 D:T7 S:rbp
OPER: addq $-8, %'d0 D:[T7] S:[T7]
OPER: movq %'s0, (%'s1) D:[] S:[rdi,T7]
OPER: leaq L1(%rip), %'d0 D:[T8] S:[]
MOVE: movq %'s0, %'d0 D:T0 S:T8
MOVE: movq %'s0, %'d0 D:rdi S:T0
MOVE: movq %'s0, %'d0 D:rsi S:rsp
OPER: call print D:[rax,r10,r11,rdi,rsi,rdx,rcx,r8,r9] S:[rdi,rsi]
OPER: movq $0, %'d0 D:[T9] S:[]
MOVE: movq %'s0, %'d0 D:rax S:T9
MOVE: movq %'s0, %'d0 D:rbx S:T2
MOVE: movq %'s0, %'d0 D:r12 S:T3
MOVE: movq %'s0, %'d0 D:r13 S:T4
MOVE: movq %'s0, %'d0 D:r14 S:T5
MOVE: movq %'s0, %'d0 D:r15 S:T6
OPER: jmp 'j0 D:[] S:[] J:[L2]
LABEL: L2
OPER:  D:[] S:[rax,rsp,rbp,rbx,r12,r13,r14,r15]
;;-END-FRAME-:
;;--FLOW--L0__tigermain_0:
def:
	n0(LABEL: L3): 
	n1(MOVE: movq %'s0, %'d0 D:T2 S:rbx): T2
	n2(MOVE: movq %'s0, %'d0 D:T3 S:r12): T3
	n3(MOVE: movq %'s0, %'d0 D:T4 S:r13): T4
	n4(MOVE: movq %'s0, %'d0 D:T5 S:r14): T5
	n5(MOVE: movq %'s0, %'d0 D:T6 S:r15): T6
	n6(MOVE: movq %'s0, %'d0 D:T7 S:rbp): T7
	n7(OPER: addq $-8, %'d0 D:[T7] S:[T7]): T7
	n8(OPER: movq %'s0, (%'s1) D:[] S:[rdi,T7]): 
	n9(OPER: leaq L1(%rip), %'d0 D:[T8] S:[]): T8
	n10(MOVE: movq %'s0, %'d0 D:T0 S:T8): T0
	n11(MOVE: movq %'s0, %'d0 D:rdi S:T0): rdi
	n12(MOVE: movq %'s0, %'d0 D:rsi S:rsp): rsi
	n13(OPER: call print D:[rax,r10,r11,rdi,rsi,rdx,rcx,r8,r9] S:[rdi,rsi]): r10, r11, r8, r9, rax, rcx, rdi, rdx, rsi
	n14(OPER: movq $0, %'d0 D:[T9] S:[]): T9
	n15(MOVE: movq %'s0, %'d0 D:rax S:T9): rax
	n16(MOVE: movq %'s0, %'d0 D:rbx S:T2): rbx
	n17(MOVE: movq %'s0, %'d0 D:r12 S:T3): r12
	n18(MOVE: movq %'s0, %'d0 D:r13 S:T4): r13
	n19(MOVE: movq %'s0, %'d0 D:r14 S:T5): r14
	n20(MOVE: movq %'s0, %'d0 D:r15 S:T6): r15
	n21(OPER: jmp 'j0 D:[] S:[] J:[L2]): 
	n22(LABEL: L2): 
	n23(OPER:  D:[] S:[rax,rsp,rbp,rbx,r12,r13,r14,r15]): 
use:
	n0(LABEL: L3): 
	n1(MOVE: movq %'s0, %'d0 D:T2 S:rbx): rbx
	n2(MOVE: movq %'s0, %'d0 D:T3 S:r12): r12
	n3(MOVE: movq %'s0, %'d0 D:T4 S:r13): r13
	n4(MOVE: movq %'s0, %'d0 D:T5 S:r14): r14
	n5(MOVE: movq %'s0, %'d0 D:T6 S:r15): r15
	n6(MOVE: movq %'s0, %'d0 D:T7 S:rbp): rbp
	n7(OPER: addq $-8, %'d0 D:[T7] S:[T7]): T7
	n8(OPER: movq %'s0, (%'s1) D:[] S:[rdi,T7]): T7, rdi
	n9(OPER: leaq L1(%rip), %'d0 D:[T8] S:[]): 
	n10(MOVE: movq %'s0, %'d0 D:T0 S:T8): T8
	n11(MOVE: movq %'s0, %'d0 D:rdi S:T0): T0
	n12(MOVE: movq %'s0, %'d0 D:rsi S:rsp): rsp
	n13(OPER: call print D:[rax,r10,r11,rdi,rsi,rdx,rcx,r8,r9] S:[rdi,rsi]): rdi, rsi
	n14(OPER: movq $0, %'d0 D:[T9] S:[]): 
	n15(MOVE: movq %'s0, %'d0 D:rax S:T9): T9
	n16(MOVE: movq %'s0, %'d0 D:rbx S:T2): T2
	n17(MOVE: movq %'s0, %'d0 D:r12 S:T3): T3
	n18(MOVE: movq %'s0, %'d0 D:r13 S:T4): T4
	n19(MOVE: movq %'s0, %'d0 D:r14 S:T5): T5
	n20(MOVE: movq %'s0, %'d0 D:r15 S:T6): T6
	n21(OPER: jmp 'j0 D:[] S:[] J:[L2]): 
	n22(LABEL: L2): 
	n23(OPER:  D:[] S:[rax,rsp,rbp,rbx,r12,r13,r14,r15]): r12, r13, r14, r15, rax, rbp, rbx, rsp
ismove:
	n0(LABEL: L3): false
	n1(MOVE: movq %'s0, %'d0 D:T2 S:rbx): true
	n2(MOVE: movq %'s0, %'d0 D:T3 S:r12): true
	n3(MOVE: movq %'s0, %'d0 D:T4 S:r13): true
	n4(MOVE: movq %'s0, %'d0 D:T5 S:r14): true
	n5(MOVE: movq %'s0, %'d0 D:T6 S:r15): true
	n6(MOVE: movq %'s0, %'d0 D:T7 S:rbp): true
	n7(OPER: addq $-8, %'d0 D:[T7] S:[T7]): false
	n8(OPER: movq %'s0, (%'s1) D:[] S:[rdi,T7]): false
	n9(OPER: leaq L1(%rip), %'d0 D:[T8] S:[]): false
	n10(MOVE: movq %'s0, %'d0 D:T0 S:T8): true
	n11(MOVE: movq %'s0, %'d0 D:rdi S:T0): true
	n12(MOVE: movq %'s0, %'d0 D:rsi S:rsp): true
	n13(OPER: call print D:[rax,r10,r11,rdi,rsi,rdx,rcx,r8,r9] S:[rdi,rsi]): false
	n14(OPER: movq $0, %'d0 D:[T9] S:[]): false
	n15(MOVE: movq %'s0, %'d0 D:rax S:T9): true
	n16(MOVE: movq %'s0, %'d0 D:rbx S:T2): true
	n17(MOVE: movq %'s0, %'d0 D:r12 S:T3): true
	n18(MOVE: movq %'s0, %'d0 D:r13 S:T4): true
	n19(MOVE: movq %'s0, %'d0 D:r14 S:T5): true
	n20(MOVE: movq %'s0, %'d0 D:r15 S:T6): true
	n21(OPER: jmp 'j0 D:[] S:[] J:[L2]): false
	n22(LABEL: L2): false
	n23(OPER:  D:[] S:[rax,rsp,rbp,rbx,r12,r13,r14,r15]): false
succesors:
	n0(LABEL: L3): n1
	n1(MOVE: movq %'s0, %'d0 D:T2 S:rbx): n2
	n2(MOVE: movq %'s0, %'d0 D:T3 S:r12): n3
	n3(MOVE: movq %'s0, %'d0 D:T4 S:r13): n4
	n4(MOVE: movq %'s0, %'d0 D:T5 S:r14): n5
	n5(MOVE: movq %'s0, %'d0 D:T6 S:r15): n6
	n6(MOVE: movq %'s0, %'d0 D:T7 S:rbp): n7
	n7(OPER: addq $-8, %'d0 D:[T7] S:[T7]): n8
	n8(OPER: movq %'s0, (%'s1) D:[] S:[rdi,T7]): n9
	n9(OPER: leaq L1(%rip), %'d0 D:[T8] S:[]): n10
	n10(MOVE: movq %'s0, %'d0 D:T0 S:T8): n11
	n11(MOVE: movq %'s0, %'d0 D:rdi S:T0): n12
	n12(MOVE: movq %'s0, %'d0 D:rsi S:rsp): n13
	n13(OPER: call print D:[rax,r10,r11,rdi,rsi,rdx,rcx,r8,r9] S:[rdi,rsi]): n14
	n14(OPER: movq $0, %'d0 D:[T9] S:[]): n15
	n15(MOVE: movq %'s0, %'d0 D:rax S:T9): n16
	n16(MOVE: movq %'s0, %'d0 D:rbx S:T2): n17
	n17(MOVE: movq %'s0, %'d0 D:r12 S:T3): n18
	n18(MOVE: movq %'s0, %'d0 D:r13 S:T4): n19
	n19(MOVE: movq %'s0, %'d0 D:r14 S:T5): n20
	n20(MOVE: movq %'s0, %'d0 D:r15 S:T6): n21
	n21(OPER: jmp 'j0 D:[] S:[] J:[L2]): n22
	n22(LABEL: L2): n23
	n23(OPER:  D:[] S:[rax,rsp,rbp,rbx,r12,r13,r14,r15]): 
;;-END-FLOW-:
yes!!
Hello WorldReturn code: 0
